### ISSUE TITLE
How can I remove the desktop icon?

### ISSUE THREAD

**Issue by davitof** [winsiderss/systeminformer#2150]
Hello. 

I recently upgraded to 3.1.24208.0. During the update, the installer created an icon on my desktop. I couIdn't find a way to remove it. 

I am using a non-admin account, and I type the admin pin each time I have to. Usually, when I try to remove an icon from my desktop (which happens with a lot of installers), Windows asks me for the admin pin and proceeds to delete it. 

But this time, after entering the pin, I got a second message telling me that I needed an authorization from Administrators to modify this file. I tried to take ownership of the file but couldn't either. It seems that Windows (successfully) edits the ownership of the linked executable instead of changing the ownership of the .lnk file. 

I tried logging as the only administrator of my laptop (which is not named Administrator), but met exactly the same problems. It seems that even an administrator must take ownership of a file before deleting it. I tried going to \users\Public\Desktop and delete or take ownership of the .lnk file, but failed there too.

How can I delete this icon? Is this behavior expected? Shouldn't all administrators have the right to delete this icon?

**Comment by davitof** [comment]
The System Informer installer does not behave exactly like other installers because although a great number use indeed the Public folder, I never ran into this problem.

**Comment by davitof** [comment]
Reopening this issue because unfortunately your answer did not work

**Comment by dmex** [comment]
> he System Informer installer does not behave exactly like other installers because although a great number use indeed the Public folder, I never ran into this problem.

 The project was created in 2008 and there's never been an issue with our shortcuts.

> How can I delete this icon? Is this behavior expected? Shouldn't all administrators have the right to delete this icon?

I have no idea. The setup sends the filename to Windows (Explorer and it decides who/what/how the shortcuts are created. 
We use the same code documented on MSDN: https://learn.microsoft.com/en-us/windows/win32/shell/links#creating-a-shortcut-and-a-folder-shortcut-to-a-file

> I tried logging as the only administrator of my laptop (which is not named Administrator), but met exactly the same problems.

Your security software or some other driver is likely interfering with the permissions.

Process Monitor would show why explorer is failing to remove the file and the stack trace of that log entry might show the driver responsible: https://learn.microsoft.com/en-us/sysinternals/downloads/procmon
or 
```fltmc instances` to review the file system filter drivers possibly blocking access .

Otherwise A) Reset and delete the file manually from an administrative command prompt.
```
icacls "\\?\C:\Users\Public\Desktop\System Informer.lnk" /reset
```
```
Remove-Item -Path "\\?\C:\Users\Public\Desktop\System Informer.lnk" -force
```

2) If you have winrar or 7zip just right-click the file and add to a zip. They have a checkbox "delete files after compression"  that remove the file.

3)  Reboot into safe mode and attempt the above a second time.

**Comment by davitof** [comment]
Sorry for not answering sooner. Thank you for your suggestions. 

Yes, of course, my security solution... I changed a few months ago (for security reasons), but I am not quite as happy with the new one, it offers much fewer ways to control it. And some things are much less user-friendly: I counted that to disable it, I would need 12 operations. And of course as many to re-enable it after. 

So I won't even try this. I like to understand and investigate, but there are limits! I'll see what ProcMon shows. And if nothing else works, your 7zip suggestion just reminded me that I there are tools to delete files upon reboot.
